File: taotao_sellergoods_interface/src/main/java/com/taotao/sellergoods/service/BrandService.java
package com.taotao.sellergoods.service;

import com.taotao.entity.PageResult;
import com.taotao.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> findAll();

    /**
     * 返回分页列表
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加
     */
    void add(TbBrand brand);

    /**
     * 回显
     */
    TbBrand findOne(Long id);

    /**
     * 更新
     */
    void update(TbBrand brand);

    /**
     * 批量删除
     */
    void delete(Long[] ids);

    /**
     * 查询
     */
    PageResult search(TbBrand brand, int pageNum, int pageSize);

    /**
     * 品牌下拉数据
     */
    List<Map> selectOptionList();
}

File: taotao_manager_web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service('brandService',function ($http) {
    this.findAll=function () {
        return $http.get('/brand/findAll.do');
    }
    this.findPage=function (pageNum,pageSize) {
        return $http.get('/brand/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize);
    }
    this.add=function (entity) {
        return $http.post('/brand/add.do',entity);
    }
    this.update=function (entity) {
        return $http.post('/brand/update.do',entity);
    }
    this.findOne=function (id) {
        return $http.get('/brand/findOne.do?id='+id);
    }
    this.dele=function (selectIds) {
        return $http.get('/brand/delete.do?ids='+selectIds);
    }
    this.search=function (pageNum,pageSize,searchEntity) {
        return $http.post('/brand/search.do?pageNum='+pageNum+'&pageSize='+pageSize,searchEntity);
    }
    this.selectBrandList=function () {
        return $http.get('/brand/selectBrandList.do');
    }
})